sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -4,6 +4,11 @@
 
 ```ts
 
+// @public
+export type AwaitedLike<T> = T extends object & {
```

Comment:
`AwaitedLike` is somewhat strange as a name since it makes it sound like an interface for something that is "like an Awaited instance." Maybe something like `UnwrapThenable` if you really don't want to overlap with TS4.5's `Awaited` name.

Since tracing is in beta we can remove this in the future if we decide that we can rely on TS4.5's definition.

Reply:
Also, since `{ then: ... }` is a subtype of `object`, there shouldn't be any difference between `object & { then: ... }` and `{ then: ... }`.

Reply:
> Also, since `{ then: ... }` is a subtype of `object`, there shouldn't be any difference between `object & { then: ... }` and `{ then: ... }`.

Thanks, yeah good call!

As far as the name... maybe `AwaitedThenable`? I like having `Awaited` in the name what do you think?

Reply:
I was chatting with Jeff and we went with `Resolved<T>` which feels appropriate here where the returntype is:

```
Promise<Resolved<ReturnType<Callback>>>;
```